kernelspec-spec/kernelspec.schema.json

```
@@ -0,0 +1,98 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "https://schema.jupyter.org/kernelspec-v1.0.schema.json",
+    "title": "Jupyter Kernelspec",
+    "description": "A description of the data required to start and manage a Jupyter Kernel",
+    "type": "object",
+
+    "definitions": {
+        "kernel_arguments": {
+            "type": "object",
+            "required": ["argv"],
+            "properties": {
+                "argv": {
+                    "description": "A list of command line arguments used to start the kernel. The text {connection_file} in any argument will be replaced with the path to the connection file.",
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    },
+                    "minItems": 3
```

I don't believe this needs to be 3, e.g. `['kernel-executable', '{connection_file}']`. I _think_ the connection file is allowed to be passed via the env and not the CLI (not certain, but this should probably be possible in the future), so I think the real minimum length is 1:

```suggestion
                    "minItems": 1
```